src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -46,13 +46,27 @@ http_get_raw_response(
     return pHttpContext->GetResponse()->GetRawHttpResponse();
 }
 
-EXTERN_C __MIDL_DECLSPEC_DLLEXPORT VOID http_set_response_status_code(
+EXTERN_C __MIDL_DECLSPEC_DLLEXPORT HRESULT http_set_response_status_code(
     _In_ IHttpContext* pHttpContext,
     _In_ USHORT statusCode,
+    _In_ USHORT subStatusCode,
     _In_ PCSTR pszReason
 )
 {
-    pHttpContext->GetResponse()->SetStatus(statusCode, pszReason);
+    HRESULT hr S_OK;
+    IN_PROCESS_STORED_CONTEXT* pInProcessStoredContext = NULL;
+    hr = IN_PROCESS_STORED_CONTEXT::GetInProcessStoredContext(pHttpContext, &pInProcessStoredContext);
+    if (FAILED(hr))
+    {
+        return hr;
+    }
+
+    if (pInProcessStoredContext->QueryRequestAborted())
```

Comment:
current design looks a lit bit wired. The request is aborted by the managed layer. Why not immediately stop reading the request at the managed layer once the managed layer decides to abort the request instead of continue reading and checking it at the native layer.

Reply:
Yeah I actually think I can remove these checks entirely. Writes and reads, if they occur, should succeed. The request will truly be aborted when the disconnect listener is fired.